src/scanpy/preprocessing/_simple.py

```diff
@@ -715,14 +756,27 @@ def regress_out(
             regres = regressors
         tasks.append(tuple((data_chunk, regres, variable_is_categorical)))
 
-    from joblib import Parallel, delayed
+    flag = None
+    if not variable_is_categorical:
+        A = regres.to_numpy()
+        # checking if inverse of A.T@A matix is zero or not.
+        if np.linalg.det(A.T @ A) != 0:
+            res = numpy_regress_out(X, A)
+            flag = 1
```

Comment:
if `flag` was necessary, it should have a speaking name and be a `bool`.

But it isn’t, you can just do `res = None` and set it in the branch, then check `if variable_is_categorical or res is None` later